ocs_ci/deployment/deployment.py

```diff
@@ -1402,6 +1431,53 @@ def deploy_with_external_mode(self):
         cluster_data = templating.load_yaml(constants.EXTERNAL_STORAGE_CLUSTER_YAML)
         cluster_data["metadata"]["name"] = config.ENV_DATA["storage_cluster_name"]
 
+        # Use Custom Storageclass Names
+        if config.ENV_DATA.get("custom_default_storageclass_names"):
+            storageclassnames = config.ENV_DATA.get("storageclassnames")
+
+            keys_to_update = [
+                constants.OCS_COMPONENTS_MAP["cephfs"],
+                constants.OCS_COMPONENTS_MAP["rgw"],
+                constants.OCS_COMPONENTS_MAP["blockpools"],
+            ]
+
+            cluster_data.setdefault("spec", {}).setdefault("managedResources", {})
+
+            for key in keys_to_update:
+                if storageclassnames.get(key):
+                    cluster_data["spec"]["managedResources"][key] = {
+                        "storageClassName": storageclassnames[key]
+                    }
+
+            # Setting up nonResilientPools custome storageclass names
+            non_resilient_pool_key = constants.OCS_COMPONENTS_MAP[
+                "cephnonresilentpools"
+            ]
+            non_resilient_pool_data = cluster_data["spec"]["managedResources"].get(
+                non_resilient_pool_key, {}
+            )
+
+            if non_resilient_pool_data.get("enable"):
+                non_resilient_pool_data = {
+                    "enable": True,
+                    "storageClassName": storageclassnames[non_resilient_pool_key],
+                }
+            cluster_data["spec"]["managedResources"][
+                non_resilient_pool_key
+            ] = non_resilient_pool_data
+
+            # Setting up custom storageclass names for 'nfs' service
+            if cluster_data["spec"].get("nfs", {}).get("enable"):
+                cluster_data["spec"]["nfs"]["storageClassName"] = storageclassnames[
+                    "nfs"
+                ]
+
+            # Setting up custom storageclass names for 'encryption' service
+            if cluster_data["spec"].get("encryption", {}).get("enable"):
+                cluster_data["spec"]["encryption"][
+                    "storageClassName"
+                ] = storageclassnames["nfs"]
```

Comment:
should be storageclassnames["encyption"]

Reply:
Done